lib/trento_web/controllers/v1/activity_log_controller.ex

```diff
@@ -11,15 +11,110 @@ defmodule TrentoWeb.V1.ActivityLogController do
   operation :get_activity_log,
     summary: "Fetches the Activity Log entries.",
     tags: ["Platform"],
+    parameters: [
+      first: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :integer},
+        required: false
+      ],
+      last: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :integer},
+        required: false
+      ],
+      after: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :string},
+        required: false
+      ],
+      before: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :string},
+        required: false
+      ],
+      from_date: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :string},
+        required: false
+      ],
+      to_date: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :string},
+        required: false
+      ],
+      actor: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :array},
+        required: false
+      ],
+      type: [
+        in: :query,
+        schema: %OpenApiSpex.Schema{type: :array},
+        required: false
+      ]
+    ],
     responses: [
       ok: {"Activity Log settings fetched successfully", "application/json", Schema.ActivityLog}
     ]
 
-  def get_activity_log(conn, _) do
-    with activity_log_entries <- ActivityLog.list_activity_log() do
+  def get_activity_log(conn, params) do
+    parsed_params =
+      case params do
+        [] -> %{}
+        params -> parse_params(params)
+      end
+
+    with {:ok, activity_log_entries, meta} <- ActivityLog.list_activity_log(parsed_params) do
       render(conn, "activity_log.json", %{
-        activity_log: activity_log_entries
+        activity_log: activity_log_entries,
+        pagination: meta
       })
     end
   end
+
+  @default_params %{
+    first: 25,
+    order_by: [:inserted_at],
+    order_direction: [:desc]
+  }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** What's the need for these defaults, as we already have defaults in the schema? (I tried to remove them and the test are still passing, though)‏

Reply:
I see one failing test with these default params set to `%{}`. Let's sync about it.